From a50bf23fca995d6fc45fd3418d0f5fb1ffc7ad24 Mon Sep 17 00:00:00 2001
Message-Id: <a50bf23fca995d6fc45fd3418d0f5fb1ffc7ad24.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Wed, 3 Dec 2014 15:29:32 +0530
Subject: [PATCH 17/28] MUST_REBASE [VPG]: adf/intel: handle panelfitter type
 in validate

In case of panel fitter we need not do a seperate validate or flip
and it will be done as part of the respective flip. We only need to
update certain flags so as to include the panel fitter changes which
is being done in the pre_validate.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-4647
Change-Id: I76422f55bb8d5c580462e82e67de1b355dba441b
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/video/adf/intel/intel_adf_device.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/adf/intel/intel_adf_device.c b/drivers/video/adf/intel/intel_adf_device.c
index 7f91a04..24728ee 100644
--- a/drivers/video/adf/intel/intel_adf_device.c
+++ b/drivers/video/adf/intel/intel_adf_device.c
@@ -378,6 +378,10 @@ static int intel_adf_device_validate(struct adf_device *dev,
 	for (i = 0; i < n_configs; i++) {
 		custom_config = &custom->configs[i];
 
+		/* In case of panel fitter dont proceed as there is no buf */
+		if (custom_config->type == INTEL_ADF_CONFIG_PANELFITTER)
+			continue;
+
 		/* verify interface id set in plane */
 		intf = idr_find(&dev->interfaces,
 			custom_config->interface_id);
-- 
1.7.9.5

